### PR TITLE
vscode configuration and output file for python build script

### PR DESCRIPTION
**Pull request by valeriof7**
Changes:
- new file:   .clang-format
- modified:   build.py
- modified:   README.md

build.py: Add option to output logs to file and update argument parsing

- Introduced a new argument `--output-to-file` to control whether logs are printed to a file or the terminal.
- Default output file is set to `build/build_output.txt`.
- Updated argument parsing to handle the new `--output-to-file` option.
- Ensured that the output is redirected to the specified file if the `--output-to-file` option is provided, otherwise logs are printed to the terminal.

README.md: Add configuration for VSCode with CMake presets and Clang-format instructions

- Added detailed instructions on how to configure VSCode using CMake presets
- Provided comments and explanations for the `CMakePresets.json` configuration
- Added guidance on selecting build and configure presets in VSCode
- Included troubleshooting tips for common issues related to CMake paths and generators
- instruction to configure and use Clang-formatter

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
-    )
-
+    print("  -h, --help            Show this help message and exit")
+    print("  -j N                  Number of processors to use")
+    print("  -c, --clean           Clean build")
+    print("  --build-dir DIR       Build directory")
+    print("  --no-tests            Do not build tests")
+    print("  --cxx-std STD         C++ standard to use")
+    print("  --build-type TYPE     Build type (e.g., Release, Debug)")
+    print("  --output-to-file      Output logs to file instead of terminal")
```
Comment (DominicDirkx) [inline build.py]: In this block, why did you remove the test on default output?

File: build.py [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
 from contextlib import chdir
-import subprocess
-import os
 
-
-def usage() -> None:
-    """Print usage information for the script."""
-
-    print("Usage: python build.py [OPTIONS]", end="\n\n")
+def usage():
```
Comment (DominicDirkx) [inline build.py:8]: Why remove the typehint on return type?

File: build.py [hunk 3]
```diff
@@ -31,6 +26,8 @@ def usage() -> None:
         "RUN_TESTS": True,
         "CXX_STANDARD": "14",
         "BUILD_TYPE": "Release",
+        "OUTPUT_TO_FILE": False,
+        "OUTPUT_FILE": "build/build_output.txt",
```
Comment (DominicDirkx) [inline build.py:36]: The build directory (build) is hard-coded here, but should be case-dependent right?

**Comment by DominicDirkx** [comment]
Thanks for adding this! And apologies for the delay on my side. I've added a few questions in the update to the build file. The update to the docs looks very good! And: thank a lot for adding the .clang-format file. I'll play around with this after it's merged and check if we need any fine-tuning on it.

File: build.py (hunk 2 above)
Comment (valeriof7) [reply build.py:8]: Just my python style, I'm not used to declare return type. You can reject without problems.

File: build.py (hunk 3 above)
Comment (valeriof7) [reply build.py:36]: This is the default value, the user can input his own path using the --output-to-file

File: build.py (hunk 1 above)
Comment (valeriof7) [reply build.py]: Sorry I don't get, what do you mean "test on default output"? If you mean specifying the default in the output message just forgot to include them

Comment (DominicDirkx) [reply build.py]: It seems that you modified the text compared to the original version. For instance:

Number of processors to use [Default: 1]

was changed to:

Number of processors to use

which provides less information

Comment (valeriof7) [reply build.py]: Fixed in commit

**Comment by DominicDirkx** [comment]
Thanks for the modifications! @alfonsoSR, since you wrote the original version, could you look over the changes in the build.py to see if you see anything that may conflict with what you did/intended: https://github.com/tudat-team/tudat-bundle/pull/25/files#diff-b70bfb027a8ef3ec4ab07a53a3b8ba83e63334ba3ce5ca6d9d773900ed452e5e

**Comment by alfonsoSR** [comment]
I'm not sure if replacing the try...finally block with a context manager is an option, but i think that is the only thing i would change

**Comment by valeriof7** [comment]
I'm not familiar with the content manager from `contextlib`, but I've tried it and don't see any difference except for more readable code (not entirely sure btw).  Here is a possible implementation:
```
@contextlib.contextmanager
def change_directory(path):
    original_directory = os.getcwd()
    os.chdir(path)
    try:
        yield
    finally:
        os.chdir(original_directory)

try:
    # Ensure build directory exists
    build_dir = Path(ARGUMENTS["BUILD_DIR"]).resolve()
    build_dir.mkdir(parents=True, exist_ok=True)

    # Build
    with change_directory(build_dir):
        outcome = subprocess.run(
            [
                "cmake",
                f"-DCMAKE_PREFIX_PATH={CONDA_PREFIX}",
                f"-DCMAKE_INSTALL_PREFIX={CONDA_PREFIX}",
                f'-DCMAKE_CXX_STANDARD={ARGUMENTS["CXX_STANDARD"]}',
                "-DBoost_NO_BOOST_CMAKE=ON",
                f'-DCMAKE_BUILD_TYPE={ARGUMENTS["BUILD_TYPE"]}',
                f'-DTUDAT_BUILD_TESTS={ARGUMENTS["BUILD_TESTS"]}',
                "..",
            ],
            stdout=output_dest,
            stderr=output_dest,
        )
        if outcome.returncode:
            exit(outcome.returncode)

        build_command = ["cmake", "--build", "."]
        if ARGUMENTS["CLEAN_BUILD"]:
            build_command.append("--target")
            build_command.append("clean")
        build_command.append(f"-j{ARGUMENTS['NUMBER_OF_PROCESSORS']}")
        outcome = subprocess.run(
            build_command, stdout=output_dest, stderr=output_dest
        )
        if outcome.returncode:
            exit(outcome.returncode)
finally:
    if output_dest:
        output_dest.close()
```

Personally, I don't think it's necessary here.

**Comment by DominicDirkx** [comment]
Since this seems to be functionally equivalent (?) I'll merge this as it is, and we can always open an issue to make corrections later.
